BlogTemplate/Models/BlogDataStore.cs

```diff
@@ -337,10 +337,19 @@ public void SaveFiles(List<IFormFile> files)
                 {                    
                     using (Stream uploadedFileStream = file.OpenReadStream())
                     {
-                        byte[] buffer = new byte[uploadedFileStream.Length];
-                        uploadedFileStream.Read(buffer, 0, buffer.Length);
                         string name = CreateFileName(file.FileName);
-                        _fileSystem.WriteFile($"{UploadsFolder}\\{name}", buffer);
+                        byte[] buffer = new byte[1024];
+                        int index = 0;
+                        while(index < buffer.Length)
+                        {
+                            int bytesRead = uploadedFileStream.Read(buffer, 0, buffer.Length - index);

```

Comment:
This looks wrong.  Don't you always want to attempt to read `buffer.Length` bytes?